src/clientiface.h

```diff
@@ -338,6 +338,7 @@ class RemoteClient
 	u8 getMajor() const { return m_version_major; }
 	u8 getMinor() const { return m_version_minor; }
 	u8 getPatch() const { return m_version_patch; }
+	std::string getFull() const { return m_full_version; }
```

Comment:
const std::string &

Reply:
getFullVersion too

Reply:
Strange it worked but:
Here it is std::string
https://github.com/minetest/minetest/blob/master/src/clientiface.h#L413

And here const std::string & 
https://github.com/minetest/minetest/blob/master/src/clientiface.h#L329

So it was wrong before. I'm not sure how to fix this :(

> getFullVersion too

Why? All other functions are also just getMinor instead of getMinorVersion

Reply:
@paramat, is it really right to use 'std::string&' here? All other code uses 'std::string'.
Looks like this is the only thing that needs to be solved to finish this PR.
Someone else opinion is needed to judge this discussion.

Reply:
- `&` prevents unnecessary string copy
- `const` signals to the compiler that some optimization can be done (I don't really know much about this optimization apart from the fact that some optimization is indeed done). It's also good practice to mark stuff that needn't be modified as `const`.

I don't see what's wrong with using `const std::string &`...

Reply:
I have little knowledge of when and where to use 'const'.

Reply:
> All other code uses 'std::string'.

May I know which functions return plain `std::string` values? `getMajor`, `getMinor`, and `getPatch` each return a `u8` value.

Reply:
It works so I mark this as resolved. If someone knows what’s exactly wrong please review again

Reply:
There's nothing wrong with the current code, but it can be _improved_ by returning a `const std::string &` instead of `std::string`. Simply change

```cpp
std::string getFull() const { return m_full_version; }
```

to

```cpp
const std::string &getFull() const { return m_full_version; }
```